### PR TITLE
Fixes deprecation in lines.py

### PR DESCRIPTION
**Pull request by rsp2210**
## PR summary
Raises runtime error on the use of non-sequence type in set_xdata and set_ydata APIs in lines.py.

## PR checklist

- [ N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: doc/api/next_api_changes/behavior/26877-RP.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+``set_xdata`` and ``set_ydata`` requires an iterable.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These APIs require argument which is iterable, use of non sequence type is deprecated.
```
Comment (tacaswell) [inline doc/api/next_api_changes/behavior/26877-RP.rst:4]:
```suggestion
These APIs require argument which is iterable, use of non sequence type is no longer supported.
```

File: lib/matplotlib/tests/test_lines.py [hunk 2]
```diff
@@ -85,22 +85,6 @@ def test_set_line_coll_dash():
...
-        mlines.Line2D([], 1)
-
-    line = mlines.Line2D([], [])
-    # when deprecation cycle is completed
-    # with pytest.raises(RuntimeError, match='x must be'):
-    with pytest.warns(mpl.MatplotlibDeprecationWarning):
-        line.set_xdata(0)
-    # with pytest.raises(RuntimeError, match='y must be'):
-    with pytest.warns(mpl.MatplotlibDeprecationWarning):
-        line.set_ydata(0)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_lines.py:101]: Please keep this test case, but change to `pytest.raises` as in the comment.

File: doc/api/next_api_changes/behavior/26877-RP.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+``set_xdata`` and ``set_ydata`` requires an iterable.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (QuLogic) [inline doc/api/next_api_changes/behavior/26877-RP.rst:2]: Please be explicit about the method location:
```suggestion
``matplotlib.lines.Line2D.set_xdata`` and ``set_ydata`` requires an iterable
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: lib/matplotlib/tests/test_lines.py [hunk 4]
```diff
@@ -85,22 +85,6 @@ def test_set_line_coll_dash():
     ax.contour(np.random.randn(20, 30), linestyles=[(0, (3, 3))])
 
 
-def test_invalid_line_data():
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_lines.py:88]: we should should not delete this test as the first half is testing a differnt code path (that we can not pass invalid inputs to the `__init__`).

File: lib/matplotlib/tests/test_lines.py [hunk 5]
```diff
@@ -85,22 +85,6 @@ def test_set_line_coll_dash():
...
-    with pytest.raises(RuntimeError, match='xdata must be'):
-        mlines.Line2D(0, [])
-    with pytest.raises(RuntimeError, match='ydata must be'):
-        mlines.Line2D([], 1)
-
-    line = mlines.Line2D([], [])
-    # when deprecation cycle is completed
-    # with pytest.raises(RuntimeError, match='x must be'):
-    with pytest.warns(mpl.MatplotlibDeprecationWarning):
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_lines.py:97]: and we should change these asserts that the calls warn to asserts that they raise (as you changed the warnings into exceptions in the library code).

**Pull request closed without merging**